backend/BackendMIIT/src/main/java/com/example/BackendMIIT/parser/AnnotationProgramsParser.java

```diff
@@ -0,0 +1,92 @@
+package com.example.BackendMIIT.parser;
+
+import com.example.BackendMIIT.model.domain.AnnotationData;
+import com.example.BackendMIIT.parser.util.ParserUtil;
+import org.jsoup.nodes.Element;
+import org.jsoup.select.Elements;
+import org.springframework.stereotype.Component;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+@Component
+public class AnnotationProgramsParser {
+
+    private static final String educationProgramsUrl = "https://www.miit.ru/sveden/education/programs";
+
+    public List<String> parseAnnotations() {
+        Map<String, AnnotationData> latestAnnotations = new HashMap<>();
+
+        try {
+            Elements rows = ParserUtil.getElements(educationProgramsUrl, "tr[itemprop='eduOp']");
+
+            for (Element row : rows) {
+                String educationForm = ParserUtil.getStringFromElement(row, "td[itemprop='eduForm']");
+                String educationLevel = ParserUtil.getStringFromElement(row, "td[itemprop='eduLevel']");
+                String profile = ParserUtil.getStringFromElement(row, "td[itemprop='eduProf']");
+
+                if (!isValidEducationLevel(educationLevel) ||
+                        !isValidEducationForm(educationForm) ||
+                        ParserUtil.isNullOrEmpty(profile)) {
+                    continue;
+                }
+
+                String annotationLink = "https://www.miit.ru" + extractAnnotationLink(row);

```

Comment:
You can set "https://www.miit.ru" as variable BASE_URL

Reply:
Moved to the application properties and used by @Value